model-lab/app/main.py

```
@@ -27,12 +32,17 @@
         raise HTTPException(status_code=400, detail="Invalid or corrupted image file") from err
-    emotion_model  = Model(model_path=MODEL_PATH, model_option=model_option)
-    emotion, prob = emotion_model.predict(pil_image=image)
-    
-    return {"emotion": emotion, "probabilities": prob}
\ No newline at end of file
+
+    try:
+        emotion_model = Model(model_path=MODEL_PATH, model_option=model_option)
+        emotion, prob = emotion_model.predict(pil_image=image)
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Model inference failed: {e}")
```

Ruff (0.14.4)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)